src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/NodeLabelMappingToURIsImpl.java

```diff
@@ -20,8 +20,8 @@ public Node map(final String label) {
     @Override
     public String unmap(final Node node) {
         if (!node.isURI())
-            throw new IllegalArgumentException("The given RDF term (" + node.toString() + ") is not an URI node.\"");
-        if (!node.getURI().startsWith(NSNODELABEL))
+            throw new IllegalArgumentException("The given RDF term (" + node.toString() + ") is not an URI node.");
+        if (!isPossibleResult(node))
```

Comment:
This new if statement also makes the previous if statement obsolete. Please remove the first if statement (and adjust the error message in this new if statement such that it mentions the two possible types of issues that the given RDF term may have).